apps/keyboard/test/unit/imes/latin/worker_test.js

```diff
@@ -443,269 +443,295 @@ suite('Latin worker', function() {
     worker = null;
   });
 
-  function onWorkerMessage(filter, callback) {
-    if (typeof filter === 'function') {
-      callback = filter;
-      filter = undefined;
-    }
-    worker.onmessage = function handleWorkerMessage(evt) {
-      if (filter !== undefined && evt.data.cmd !== filter) {
-        return;
-      }
-
-      worker.onmessage = null;
-      callback(evt.data);
-    };
+  function onWorkerMessage(filter) {
+    return new Promise(function(resolve,reject) {
+      worker.onmessage = function handleWorkerMessage(evt) {
+        if (filter !== undefined && evt.data.cmd !== filter) {
+          return;
+        }
+
+        worker.onmessage = null;
+        resolve(evt.data);
+      };
+    });
   }
 
-  function prediction(input, expected, next) {
-    worker.postMessage({cmd: 'predict', args: [input]});
+  function prediction(input, expected) {
+    return new Promise(function(resolve,reject) {
```

Comment:
Doing so also ensure any assertion thrown will correctly flow to the `done` function. The current patch will simply timeout when these assertion throws, since `resolve()` will never be called.